x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/tag_display.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiPopover, EuiBadgeGroup, EuiBadge, EuiButtonEmpty } from '@elastic/eui';
+import styled from 'styled-components';
+import * as i18n from '../translations';
+
+interface TagsDisplayProps {
+  tags: string[];
+}
+
+const TagWrapper = styled(EuiBadgeGroup)`
+  width: 100%;
+`;
+
+const TagPopoverWrapper = styled(EuiBadgeGroup)`
+  max-height: 200px;
+  max-width: 600px;
+  overflow: auto;
+`;
+
+const TagPopoverButton = styled(EuiButtonEmpty)`
+  font-size: ${({ theme }) => theme.eui.euiFontSizeXS}
+  font-weight: 500;
+  height: 20px;
+`;
+
+/**
+ * @param tags to display for filtering
+ */
+const TagsDisplayComponent = ({ tags }: TagsDisplayProps) => {
+  const [isTagPopoverOpen, setIsTagPopoverOpen] = useState(false);
+
+  return (
+    <>
+      {tags.length <= 3 ? (
+        <TagWrapper data-test-subj="tags">
+          {tags.map((tag: string, i: number) => (
+            <EuiBadge
+              color="hollow"
+              key={`${tag}-${i}`}
+              data-test-subj={`rules-table-column-tags-${i}`}
+            >
+              {tag}
+            </EuiBadge>
+          ))}
+        </TagWrapper>
+      ) : (
+        <TagWrapper data-test-subj="tags">
+          {tags.slice(0, 3).map((tag: string, i: number) => (
+            <EuiBadge
+              color="hollow"
+              key={`${tag}-${i}`}
+              data-test-subj={`rules-table-column-tags-${i}`}
+            >
+              {tag}
+            </EuiBadge>
+          ))}
+
+          <EuiPopover
+            ownFocus
+            display="block"
+            data-test-subj="tags-display-popover"
+            button={
+              <TagPopoverButton
+                size="xs"
+                data-test-subj={'tags-display-popover-button'}
+                onClick={() => setIsTagPopoverOpen(!isTagPopoverOpen)}
+              >
+                {i18n.COLUMN_SEE_ALL_POPOVER}
+              </TagPopoverButton>
+            }
+            isOpen={isTagPopoverOpen}
+            closePopover={() => setIsTagPopoverOpen(!isTagPopoverOpen)}
+            repositionOnScroll
+          >
+            <TagPopoverWrapper>
+              {tags.map((tag: string, i: number) => (
+                <EuiBadge
+                  color="hollow"
+                  key={`${tag}-${i}`}
+                  data-test-subj={`rules-table-column-popover-tags-${i}`}
+                >
```

Comment:
Interestingly enough, it doesn't seem like the `title` hover works within the popover. Adding an explicit `title` prop doesn't seem to resolve this (it's added by EUI anyway 🤷‍♂️) . Not a big deal either way, just commenting :)


<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/94965577-fffa0100-04b8-11eb-9a57-56c6ff77f050.gif" />
</p>

Reply:
I added a `title` prop and it seemed to work for me. I went ahead and added it explicitly